### PR TITLE
Fix crash in recording (didn't happen in macOS)

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #4996

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/ProjectAudioManager.cpp [hunk 1]
```diff
@@ -904,6 +904,7 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
                   t1 = newTrack->LongSamplesToTime(newTrack->TimeToLongSamples(t1));
             }
 
+            auto tempList = TrackList::Temporary(nullptr, newTrack, nullptr);
```
Comment (crsib) [inline src/ProjectAudioManager.cpp:907]: I've tried to say during the standup "Oh, wait, I saw the fix in a different PR", but I was muted.

So...

Oh, wait, I saw the fix in a different PR :-)

**Review by crsib — APPROVED** [review]

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @Paul-Licameli 's latest branch fix for this:  audacity-win-3.4.0-alpha-20230811+2294b4e-x64

First I get this:
image

Then after pressing the OK button I get:
image

Then after pressing the OK or Cancel buttons Audacity crashes

**Comment by Paul-Licameli** [comment]
> Testing on W10 with @Paul-Licameli 's latest branch fix for this: audacity-win-3.4.0-alpha-20230811+2294b4e-x64
> 
> First I get this: image
> 
> Then after pressing the OK button I get: image
> 
> Then after pressing the OK or Cancel buttons Audacity crashes
> 
> I am very curious as to why the error message refers to an issue at **D:\a\audacity ...**
> 
> I do happen to have a D: drive on this particular PC - but it is my onboard 1TB spinning metal drive.
> 
> Audacity (including this branch) is always installed on the SSD 250GB C: drive .
> 
> And all my active projects and temp files for audacity and audacity settings folder are on the C:drive too.

This happens only when recording to stereo, not to mono?

I pushed another commit that I think will complete the fix.

**Comment by petersampsonaudacity** [comment]
> This happens only when recording to stereo, not to mono?

No it also crashes when I record in Mono - but with mono I get an  error message but it is immediately overwritten with the debug report, the red recording cursor dances along - and then just crashes with no further input from me.

Shift+Record also crashes.

_I'll try the fresh commit/build later._

**Comment by petersampsonaudacity** [comment]
Testing with @Paul-Licameli 's latest branch build:  audacity-win-3.4.0-alpha-20230812+6ad372f-x64

This now appears to work fine on W10 - tested

1. stereo and mono
2. Play button
3. R shortcut key
4. Transport > Recording Record command
5. Shift modified recording
6. Timer Record

All work fine

**Review by crsib — APPROVED** [review]

**Pull request merged**